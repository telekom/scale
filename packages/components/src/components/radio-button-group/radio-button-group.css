:host {
  /*label*/
  --font-weight-label: var(--scl-font-weight-medium);
  --font-size-label: var(--scl-font-size-16);

  /*helper-text*/
  --font-size-helper-text: var(--scl-font-size-12);
  --color-helper-text: var(--scl-color-blue-70);
  --padding-bottom-helper-text: var(--scl-spacing-8);
  --margin-top-helper-text: var(--scl-spacing-4);
  --font-weight-helper-text: var(--scl-font-weight-medium);

  /*helper-text--status-error*/
  --font-size-error-helper-text: var(--scl-font-size-12);
  --color-error-helper-text: var(--scl-color-red-60);
  --padding-bottom-error-helper-text: var(--scl-spacing-8);
  --margin-top-error-helper-text: var(--scl-spacing-4);
  --font-weight-error-helper-text: var(--scl-font-weight-medium);

  /*title*/
  --font-size-title: var(--font-size-label);
  --font-weight-title: var(--scl-font-weight-medium);
  --padding-bottom-title: var(--scl-spacing-4);
  --margin-left-title: var(--scl-spacing-2);

  /*slotted items*/
  --margin-top-slotted-item: var(--scl-spacing-4);
}

.radio-button-group {
  display: inline-flex;
  flex-direction: column;
  border: 0;
}

.radio-button-group__container {
  display: flex;
  flex-direction: column;
}

.radio-button-group__helper-text {
  color: var(--color-helper-text);
  font-size: var(--font-size-helper-text);
  margin-top: calc(var(--margin-top-helper-text) * -1);
  padding-bottom: calc(var(--padding-bottom-helper-text) * -1);
  font-weight: var(--font-weight-helper-text);
}

.radio-button-group__helper-text--status-error {
  color: var(--color-error-helper-text);
  font-size: var(--font-size-error-helper-text);
  margin-top: calc(var(--margin-top-error-helper-text) * -1);
  padding-bottom: calc(var(--padding-bottom-error-helper-text) * -1);
  font-weight: var(--font-weight-error-helper-text);
}

.radio-button-group__title {
  font-size: var(--font-size-title);
  font-weight: var(--font-weight-title);
  padding-bottom: var(--padding-bottom-title);
  margin-left: calc(var(--margin-left-title) * -1.5);
}

::slotted(*) {
  margin-top: var(--margin-top-slotted-item);
}
