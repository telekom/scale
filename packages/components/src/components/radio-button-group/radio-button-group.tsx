import { Component, Element, h, Prop } from '@stencil/core';
import classNames from 'classnames';
import statusNote from '../../utils/status-note';

@Component({
  tag: 'scale-radio-button-group',
  styleUrl: './radio-button-group.css',
  shadow: true,
})
export class RadioButtonGroup {
  @Element() hostElement: HTMLElement;
  /** (optional) Input label */
  @Prop() label: string = '';
  /** (optional) Input helper text */
  @Prop() helperText?: string = '';
  /** @deprecated - invalid should replace status */
  @Prop() status?: string = '';
  /** (optional) Input status */
  @Prop() invalid?: boolean = false;

  componentDidRender() {
    if (this.status !== '') {
      statusNote({
        tag: 'deprecated',
        message:
          'Property "status" is deprecated. Please use the "invalid" property!',
        type: 'warn',
        source: this.hostElement,
      });
    }
  }

  render() {
    return (
      <fieldset class="radio-button-group">
        <legend class="radio-button-group__title">
          <label
            class="radio-button-group__title-label"
            aria-label={this.label}
          >
            {this.label}
          </label>
          {this.helperText ? (
            <div
              role="text"
              class={this.getCssClassMap()}
              aria-label={this.helperText}
            >
              {this.helperText}
            </div>
          ) : null}
        </legend>
        <div class="radio-button-group__container">
          <slot />
        </div>
      </fieldset>
    );
  }

  getCssClassMap() {
    return classNames(
      'radio-button-group__helper-text',
      (this.status === 'error' || this.invalid) &&
        `radio-button-group__helper-text--status-error`
    );
  }
}
