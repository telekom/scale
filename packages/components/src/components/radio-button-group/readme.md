# scale-radio-button-group



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description                                | Type      | Default |
| ------------ | ------------- | ------------------------------------------ | --------- | ------- |
| `helperText` | `helper-text` | (optional) Input helper text               | `string`  | `''`    |
| `invalid`    | `invalid`     | (optional) Input status                    | `boolean` | `false` |
| `label`      | `label`       | (optional) Input label                     | `string`  | `''`    |
| `status`     | `status`      | DEPRECATED - invalid should replace status | `string`  | `''`    |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
