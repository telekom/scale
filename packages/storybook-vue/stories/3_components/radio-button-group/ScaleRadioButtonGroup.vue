<template>
  <scale-radio-button-group>
    <slot></slot>
  </scale-radio-button-group>
</template>

<script>
import { action } from '@storybook/addon-actions';

export default {
  props: {
    helperText: { type: String },
    label: { type: String },
    status: { type: String },
  },
  methods: {},
};
</script>
