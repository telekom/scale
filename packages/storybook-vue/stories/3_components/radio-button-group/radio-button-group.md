<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Radio Button Group</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>

This component contains the Scale radio button group. This is used to display the scale radio buttons in a list with a suitable heading.

The heading can be extended with a helper text. If the status of the radio button group is set to error, both the help text and the circles of the radio buttons in the list change colour.

## Beta components

This component is still in the beta phase. When testing it, keep in mind that it may not have gone through all quality control measures, and it may not yet have WCAG accessibility certification. There may be changes to this component in the future.
