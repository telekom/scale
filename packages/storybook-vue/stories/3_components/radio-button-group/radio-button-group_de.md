<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Radio Button Group</h1>
    <img src="assets/tag-beta.svg" alt="Beta Component" />
</div>
Radio-Buttons ermöglichen es Nutzer*innen, eine Option aus einer Reihe von Alternativen auszuwählen. Die Radio Button Group enthält zusätzlich einen Gruppentitel.

## Allgemein

Radio-Buttons funktionieren nur als Gruppe. Durch Aktivieren eines Radio-Buttons wird die zuvor gewählte Option automatisch abgewählt. Durch einen Gruppentitel erfassen Nutzer*innen inhaltliche Zusammenhänge schneller. Interfaces mit vielen Auswahloptionen wirken insgesamt ruhiger und aufgeräumter.

![Radio Button Group](./img/Radio_Button_Group.png)

### Verwendung

#### Verwende eine Radio Button Group, um

- Nutzer*innen eine Reihe von verwandten, aber sich gegenseitig ausschließenden Auswahlmöglichkeiten anzubieten und um
- bei mehreren Gruppen von Radiobuttons die Übersichtlichkeit zu gewährleisten.

#### Verwende Radio-Buttons grundsätzlich nicht, wenn

### Don\'t use a radio button group when:

- Nutzer*innen mehr als eine Option auswählen sollen (nutze hierfür Checkboxen) oder
- Nutzer*innen eine Aktion ausführen müssen (nutze hierfür einen Button).

## Elemente

![Elemente der Radio Button Group](./img/Radio_Button_Group_Elements.png)

### Gruppenlabel (1)

Wähle einen kurzen und aussagekräftigen Titel für die gesamte Gruppe.


### Radio-Button ausgeschaltet (2)

Wenn der Kreis leer ist, gilt eine Option als abgewählt.

### Label (3)

Beschreibe mit dem Label deutlich, welche Auswirkung die jeweilige Auswahl hat. Ein Label enthält in der Regel maximal drei Wörter.


### Radio-Button eingeschaltet

Ist der Kreis ausgefüllt, so gilt eine Option als ausgewählt.

### Information (5) (optional)

Wenn nötig, kannst du auch noch eine zusätzliche Information angeben.

### Fehlermeldung (6)

Fehlermeldungen sind unterhalb des Gruppenlabels platziert.

### Deaktivierter Zustand

Deaktiviere einen Radio Button, wenn

- eine andere Aktion erfolgen muss, bevor der Radio-Button nutzbar ist oder
- die Option aufgeführt werden muss, aber aktuell nicht auswählbar ist.


![Deaktivierte Radio Button Group](./img/Radio_Button_Group_Disabled.png)

## Verwandte Komponenten

[Radio Button](?path=/usage/components-radio-button), [Checkbox](?path=/usage/components-checkbox), [Checkbox Group](?path=/usage/components-checkbox-group)
