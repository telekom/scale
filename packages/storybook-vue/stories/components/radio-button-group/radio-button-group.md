<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Radio Button Group</h1>
    <img src="assets/aa.png" alt="Accessible AA" />
</div>

Radio buttons allow users to choose an option from a number of
alternatives. The radio button group also contains a group label.

## General

Radio buttons only work as a group. Activating a radio button
automatically deselects the previously selected option. With a group
label, users can grasp contextual context more quickly. Interfaces with
many options appear smoother and tidier overall.

![Radio Button Group](assets/3_components/radio-button-group/Radio_Button_Group.png)

### When to use

Use a radio button group to:

- offer users a range of related but mutually exclusive choices and
- to ensure clarity in the case of several groups of radio buttons.

#### When not to use

Don't use a radio button group when:

- users should select more than one option (use checkboxes instead) or
- users have to perform an action (use a button instead).

## Elements

![Elements of Radio Button Group](assets/3_components/radio-button-group/Radio_Button_Group_Elements.png)

### Group label (1)

Choose a short and meaningful title for the entire group.

### Deselected radio button (2)

If the circle is empty, an option is deselected.

### Label (3)

Use the label to clearly describe the effect of the respective selection. A label usually contains max three words.

### Selected radio button

If the circle is filled in, an option is selected.

### Information (5) (optional)

If necessary, you can also provide additional information.

### Error message (6)

Place error messages below the group label.

## Deactivated state

Deactivate a radio button when:

- users have to complete another action before selecting the radio
  button or
- the option must be listed but users can\'t currently select it.

![Disabled Radio Button Group](assets/3_components/radio-button-group/Radio_Button_Group_Disabled.png)

## Related components

[Radio Button](?path=/usage/components-radio-button), [Checkbox](?path=/usage/components-checkbox), [Checkbox Group](?path=/usage/components-checkbox-group)
