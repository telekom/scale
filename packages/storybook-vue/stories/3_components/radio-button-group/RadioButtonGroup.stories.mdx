import {
  Meta,
  Story,
  ArgsTable,
  Canvas,
  Description,
} from '@storybook/addon-docs/blocks';
import ScaleRadioButtonGroup from './ScaleRadioButtonGroup.vue';
import { action } from '@storybook/addon-actions';

<Meta
  title="Beta Components/Radio Button Group"
  component={ScaleRadioButtonGroup}
  argTypes={{}}
/>

export const StandardTemplate = (args, { argTypes }) => ({
  props: {
    width: { type: String },
    ...ScaleRadioButtonGroup.props,
  },
  computed: {
    wrapperStyles() {
      return {
        maxWidth: this.width,
      };
    },
  },
  template: `
    <div :style="wrapperStyles">
    <scale-radio-button-group :label="label" :status="status" :helper-text="helperText">
      <scale-radio-button
        value="1"
        label="Radio Button"
        name="radioStandard"
        transparent="true"
        input-id="choiceStandardOne"
      ></scale-radio-button>
      <scale-radio-button
        value="2"
        label="Radio Button"
        name="radioStandard"
        input-id="choiceStandardTwo"
      ></scale-radio-button>
    </scale-radio-button-group>
    </div>
  `,
});

export const HelperTextTemplate = (args, { argTypes }) => ({
  props: {
    width: { type: String },
    ...ScaleRadioButtonGroup.props,
  },
  computed: {
    wrapperStyles() {
      return {
        maxWidth: this.width,
      };
    },
  },
  template: `
    <div :style="wrapperStyles">
    <scale-radio-button-group :label="label" :status="status" :helper-text="helperText">
      <scale-radio-button
        value="1"
        label="Radio Button"
        name="RadioButtonHelperText"
        transparent="true"
        input-id="choiceHelperTextOne"
      ></scale-radio-button>
      <scale-radio-button
        value="2"
        label="Radio Button"
        name="RadioButtonHelperText"
        input-id="choiceHelperTextTwo"
      ></scale-radio-button>
    </scale-radio-button-group>
    </div>
  `,
});

export const ErrorTemplate = (args, { argTypes }) => ({
  props: {
    width: { type: String },
    ...ScaleRadioButtonGroup.props,
  },
  computed: {
    wrapperStyles() {
      return {
        maxWidth: this.width,
      };
    },
  },
  template: `
    <div :style="wrapperStyles">
    <scale-radio-button-group :label="label" :status="status" :helper-text="helperText">
      <scale-radio-button
        status="error"
        value="1"
        label="Radio Button"
        name="radioButtonError"
        transparent="true"
        input-id="choiceErrorOne"
      ></scale-radio-button>
      <scale-radio-button
        status="error"
        value="2"
        label="Radio Button"
        name="radioButtonError"
        input-id="choiceErrorTwo"
      ></scale-radio-button>
    </scale-radio-button-group>
    </div>
  `,
});

## Standard

<Canvas withSource="none">
  <Story
    name="Standard"
    args={{
      label: 'Group Label',
    }}
  >
    {StandardTemplate.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Standard" />

```html
<scale-radio-button-group label="Group Label">
  <scale-radio-button
    value="1"
    label="Radio Button"
    name="radioStandard"
    transparent="true"
    input-id="choiceStandardOne"
  ></scale-radio-button>
  <scale-radio-button
    value="2"
    label="Radio Button"
    name="radioStandard"
    input-id="choiceStandardTwo"
  ></scale-radio-button>
</scale-radio-button-group>
```

## Helper Text

<Canvas withSource="none">
  <Story
    name="Helper Text"
    args={{
      label: 'Group Label',
      helperText: 'Helper message',
    }}
  >
    {HelperTextTemplate.bind({})}
  </Story>
</Canvas>

```html
<scale-radio-button-group label="Group Label" helper-text="Helper message">
  <scale-radio-button
    value="1"
    label="Radio Button"
    name="RadioButtonHelperText"
    transparent="true"
    input-id="choiceHelperTextOne"
  ></scale-radio-button>
  <scale-radio-button
    value="2"
    label="Radio Button"
    name="RadioButtonHelperText"
    input-id="choiceHelperTextTwo"
  ></scale-radio-button>
</scale-radio-button-group>
```

## Error

<Canvas withSource="none">
  <Story
    name="Error"
    args={{
      label: 'RadioButton Group',
      status: 'error',
      helperText: 'Error message',
    }}
  >
    {ErrorTemplate.bind({})}
  </Story>
</Canvas>

```html
<scale-radio-button-group
  label="RadioButton Group"
  status="error"
  helper-text="Error message"
>
  <scale-radio-button
    status="error"
    value="1"
    label="Radio Button"
    name="radioButtonError"
    transparent="true"
    input-id="choiceErrorOne"
  ></scale-radio-button>
  <scale-radio-button
    status="error"
    value="2"
    label="Radio Button"
    name="radioButtonError"
    input-id="choiceErrorTwo"
  ></scale-radio-button>
</scale-radio-button-group>
```
