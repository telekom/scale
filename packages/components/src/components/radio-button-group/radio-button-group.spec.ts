import { newSpecPage } from '@stencil/core/testing';
import { RadioButtonGroup } from './radio-button-group';

describe('RadioButtonGroup', () => {
  it('should match standard snapshot', async () => {
    const page = await newSpecPage({
      components: [RadioButtonGroup],
      html: `
              <scale-radio-button-group label="Radio Button Label">
              <div slot="radio-button-list">
                <scale-radio-button
                  value="1"
                  label="Radio Button"
                  name="radioSelect"
                  transparent="true"
                  input-id="choiceOne"
                ></scale-radio-button>
                <div style="margin-top: 0.5rem"></div>
                <scale-radio-button
                  value="2"
                  label="Radio Button"
                  name="radioSelect"
                  input-id="choiceTwo"
                ></scale-radio-button>
                <div style="margin-top: 0.5rem"></div>
                <scale-radio-button
                  value="3"
                  label="Radio Button"
                  name="radioSelect"
                  input-id="choiceThree"
                ></scale-radio-button>
              </div>
            </scale-radio-button-group>`,
    });
    expect(page.root).toMatchSnapshot();
  });
  it('should match selected snapshot', async () => {
    const page = await newSpecPage({
      components: [RadioButtonGroup],
      html: `
      <scale-radio-button-group label="Radio Button Label">
      <div slot="radio-button-list">
        <scale-radio-button
          value="1"
          label="Radio Button"
          name="radioSelectedOne"
          transparent="true"
          checked
          input-id="choiceOne"
        ></scale-radio-button>
        <div style="margin-top: 0.5rem"></div>
        <scale-radio-button
          value="2"
          label="Radio Button"
          name="radioSelectedTwo"
          input-id="choiceTwo"
        ></scale-radio-button>
        <div style="margin-top: 0.5rem"></div>
        <scale-radio-button
          value="3"
          label="Radio Button"
          name="radioSelectedThree"
          input-id="choiceThree"
        ></scale-radio-button>
      </div>
    </scale-radio-button-group>
`,
    });
    expect(page.root).toMatchSnapshot();
  });
  it('should match error snapshot', async () => {
    const page = await newSpecPage({
      components: [RadioButtonGroup],
      html: `
      <scale-radio-button-group
        label="Radio Button Label"
        helper-text="Error message"
        >
        <div slot="radio-button-list">
        <scale-radio-button
          value="1"
          status="error"
          label="Radio Button"
          name="radioError"
          transparent="true"
          input-id="choiceOne"
        ></scale-radio-button>
        <div style="margin-top: 0.5rem"></div>
        <scale-radio-button
          value="2"
          status="error"
          label="Radio Button"
          name="radioError"
          input-id="choiceTwo"
        ></scale-radio-button>
        <div style="margin-top: 0.5rem"></div>
        <scale-radio-button
          value="3"
          status="error"
          label="Radio Button"
          name="radioError"
          input-id="choiceThree"
        ></scale-radio-button>
      </div>
    </scale-radio-button-group>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
