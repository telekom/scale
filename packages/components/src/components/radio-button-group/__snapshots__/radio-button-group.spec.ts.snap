// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup should match error snapshot 1`] = `
<scale-radio-button-group helper-text="Error message" invalid="true" label="Radio Button Label">
  <mock:shadow-root>
    <form>
      <fieldset class="radio-button-group">
        <legend class="radio-button-group__title">
          <label aria-label="Radio Button Label" class="radio-button-group__title-label" role="text">
            Radio Button Label
          </label>
          <div aria-label="Error message" class="radio-button-group__helper-text radio-button-group__helper-text--status-error" role="text">
            Error message
          </div>
        </legend>
        <div class="radio-button-group__container">
          <slot></slot>
        </div>
      </fieldset>
    </form>
  </mock:shadow-root>
  <div slot="radio-button-list">
    <scale-radio-button input-id="choiceOne" invalid="true" label="Radio Button" name="radioError" transparent="true" value="1"></scale-radio-button>
    <div style="margin-top: 0.5rem;"></div>
    <scale-radio-button input-id="choiceTwo" invalid="true" label="Radio Button" name="radioError" value="2"></scale-radio-button>
    <div style="margin-top: 0.5rem;"></div>
    <scale-radio-button input-id="choiceThree" invalid="true" label="Radio Button" name="radioError" value="3"></scale-radio-button>
  </div>
</scale-radio-button-group>
`;

exports[`RadioButtonGroup should match selected snapshot 1`] = `
<scale-radio-button-group label="Radio Button Label">
  <mock:shadow-root>
    <form>
      <fieldset class="radio-button-group">
        <legend class="radio-button-group__title">
          <label aria-label="Radio Button Label" class="radio-button-group__title-label" role="text">
            Radio Button Label
          </label>
        </legend>
        <div class="radio-button-group__container">
          <slot></slot>
        </div>
      </fieldset>
    </form>
  </mock:shadow-root>
  <div slot="radio-button-list">
    <scale-radio-button checked="" input-id="choiceOne" label="Radio Button" name="radioSelectedOne" transparent="true" value="1"></scale-radio-button>
    <div style="margin-top: 0.5rem;"></div>
    <scale-radio-button input-id="choiceTwo" label="Radio Button" name="radioSelectedTwo" value="2"></scale-radio-button>
    <div style="margin-top: 0.5rem;"></div>
    <scale-radio-button input-id="choiceThree" label="Radio Button" name="radioSelectedThree" value="3"></scale-radio-button>
  </div>
</scale-radio-button-group>
`;

exports[`RadioButtonGroup should match standard snapshot 1`] = `
<scale-radio-button-group label="Radio Button Label">
  <mock:shadow-root>
    <form>
      <fieldset class="radio-button-group">
        <legend class="radio-button-group__title">
          <label aria-label="Radio Button Label" class="radio-button-group__title-label" role="text">
            Radio Button Label
          </label>
        </legend>
        <div class="radio-button-group__container">
          <slot></slot>
        </div>
      </fieldset>
    </form>
  </mock:shadow-root>
  <div slot="radio-button-list">
    <scale-radio-button input-id="choiceOne" label="Radio Button" name="radioSelect" transparent="true" value="1"></scale-radio-button>
    <div style="margin-top: 0.5rem;"></div>
    <scale-radio-button input-id="choiceTwo" label="Radio Button" name="radioSelect" value="2"></scale-radio-button>
    <div style="margin-top: 0.5rem;"></div>
    <scale-radio-button input-id="choiceThree" label="Radio Button" name="radioSelect" value="3"></scale-radio-button>
  </div>
</scale-radio-button-group>
`;
